src/main/java/taplink/network/menu/api/controllers/ItemController.java

```diff
@@ -0,0 +1,69 @@
+package taplink.network.menu.api.controllers;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import taplink.network.menu.api.commons.constants.AppConstants;
+import taplink.network.menu.api.dtos.request.ItemRequestDto;
+import taplink.network.menu.api.dtos.response.ItemResponseDto;
+import taplink.network.menu.api.dtos.response.ResponseDto;
+import taplink.network.menu.api.services.ItemService;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/items")
+public class ItemController {
+
+    private final ItemService itemService;
+
+    @GetMapping
+    public ResponseEntity<?> searchItems(
+            @RequestParam(value = "searchKey", defaultValue = AppConstants.EMPTY, required = false) String searchKey,
+            @RequestParam(value = "pageNo", defaultValue = AppConstants.DEFAULT_PAGE_NUMBER, required = false) int pageNo,
+            @RequestParam(value = "pageSize", defaultValue = AppConstants.DEFAULT_PAGE_SIZE, required = false) int pageSize,
+            @RequestParam(value = "sortBy", defaultValue = AppConstants.DEFAULT_SORT_BY, required = false) String sortBy,
+            @RequestParam(value = "sortDir", defaultValue = AppConstants.DEFAULT_SORT_DIRECTION, required = false) String sortDir
+    ) {
+        ResponseDto<ItemResponseDto> responseDTO = itemService.searchItems(searchKey, pageNo, pageSize, sortBy, sortDir);
+        return new ResponseEntity<>(responseDTO, HttpStatus.OK);
```

Comment:
Should not return all items, just return items that belong to a particular category.